src/lib/components/VideoEmbed.svelte

```diff
@@ -1,9 +1,21 @@
 <script lang="ts">
 	import FrameEmbed from '$lib/components/FrameEmbed.svelte'
 	import Splash from '$lib/components/Splash.svelte'
+	import { VideoSource } from '$lib/data'
 	import type { Video } from '$lib/data'
+	import { preferredSource } from '$lib/store.js'
 
 	export let video: Video
+
+	let videoSource: VideoSource
+
+	preferredSource.subscribe((value: VideoSource) => {
+		videoSource = value
+	})
+
+	function setSource(source: VideoSource) {
+		preferredSource.update(() => source)
```

Comment:
This can be simplified to `preferredSource.set(source)`

Reply:
It is done.